### DISCUSSION TITLE
Should `AnyString` translate into accepting any type that implments the `Display` trait?

### DISCUSSION THREAD

**Discussion by Supermath101** [CarlKCarlK/anyinput#4] [Ideas]
Meaning, should this:
```rs
#[anyinput]
fn len_plus_2(s: AnyString) -> usize {
    s.len() + 2
}
```
translate into this?:
```rs
fn len_plus_2(s: impl ::std::fmt::Display) -> usize {
    let s = ::std::format!("{}", s);
    s.len() + 2
}
```

Edit: Here's another possible macro expansion:
```rs
fn len_plus_2(s: impl ::std::convert::Into<String>) -> usize {
    s.into().len() + 2 // notice the `.into()` in-between.
}
```

**Comment by CarlKCarlK** [comment]
Thanks! The Display one is very fun. It could be called something like "AnyDisplay" or "AnyDisplayable" and would let you call
`println!("{}", len_plus_2(1000));`
and get an answer of 6.

I'll keep an eye out for demand for this.
